packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -28,22 +29,44 @@ class StopwatchFactory {
 typedef VoidCallback = void Function();
 
 abstract class Logger {
+  /// Whether or not this logger should print [printTrace] messages.
   bool get isVerbose => false;
 
+  /// If true, silences the logger output.
   bool quiet = false;
 
+  /// If true, this logger supports color output.
   bool get supportsColor;
 
+  /// If true, this logger is connected to a terminal.
   bool get hasTerminal;
 
+  /// If true, then [printError] has been called at least once for this logger
+  /// since the last time it was set to false.
+  bool hadErrorOutput = false;
+
+  /// Causes [checkForFatalLogs] to call [throwToolExit] when it is called if
+  /// [hadErrorOutput] is true.
+  bool errorsAreFatal = false;
+
+  /// If true, then [printWarning] has been called at least once for this logger
+  /// since the last time it was reset to false.
+  bool hadWarningOutput = false;
```

Comment:
FWIW we can make the analyzer save us by having `reportError` return `Never` so that every implementation must always throw and every call site must always assume it throws.

Like I said, I don't have a strong opinion here. We could also just check that if printError has been called, the tool doesn't exit cleanly. It doesn't have to be on a timer or anything. We could also just do the change you're describing but over a longer period of time, not in this PR.

Reply:
Ahh, `Never` was what I was missing.  That would work.